x/gamm/pool-models/stableswap/amm_test.go

```diff
@@ -19,42 +19,70 @@ func TestCFMMInvariantTwoAssets(t *testing.T) {
 	kErrTolerance := sdk.OneDec()
 
 	tests := []struct {
-		xReserve sdk.Dec
-		yReserve sdk.Dec
-		yIn      sdk.Dec
+		xReserve    sdk.Dec
+		yReserve    sdk.Dec
+		yIn         sdk.Dec
+		expectPanic bool
 	}{
 		{
 			sdk.NewDec(100),
 			sdk.NewDec(100),
 			sdk.NewDec(1),
+			false,
 		},
 		{
 			sdk.NewDec(100),
 			sdk.NewDec(100),
 			sdk.NewDec(1000),
+			false,
 		},
 		// {
 		// 	sdk.NewDec(100000),
 		// 	sdk.NewDec(100000),
 		// 	sdk.NewDec(10000),
 		// },
+
+		// panic catching
+		{ // xReserve negative
```

Comment:
nit: consider adding names to test cases

Reply:
We're going to do a proper test refactor once certain core PRs are merged as right now changing these would break every open stableswap PR, but yes these tests will eventually be changed (test and field names, map structure, test suite vars for reused cases etc.)

Reply:
Okay I just went ahead and refactored it in this one, please see recent commit